its/src/test/java/com/sonar/it/csharp/NoSourcesTest.java

```diff
@@ -0,0 +1,50 @@
+package com.sonar.it.csharp;
+
+import com.sonar.it.shared.TestUtils;
+import com.sonar.orchestrator.Orchestrator;
+import com.sonar.orchestrator.build.ScannerForMSBuild;
+import java.nio.file.Path;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import static com.sonar.it.csharp.Tests.ORCHESTRATOR;
+import static com.sonar.it.csharp.Tests.getMeasureAsInt;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class NoSourcesTest {
+
+  @ClassRule
+  public static TemporaryFolder temp = TestUtils.createTempFolder();
+
+  private static final String PROJECT = "ProjectWithNoSources";
+  @ClassRule
+  public static final Orchestrator orchestrator = Tests.ORCHESTRATOR;
+
+  @BeforeClass
+  public static void init() throws Exception {
+    TestUtils.reset(orchestrator);
+
+    Path projectDir = Tests.projectDir(temp, "ProjectWithNoSources");
```

Comment:
This initialization seems to be copy/pasted several times, can it be factored out to some `TestHelper` class?
To clarify: Only for simple cases with single `projectDir` argument.

Reply:
well, the `temp` needs to be passed from here as it's `ClassRule` annotated; and the projectDir needs to be passed... 

it's already in the `TestHelper`. 

Do you mean initializing _and_ doing the analysis? I think the way it is now makes the test understandable, each step

Reply:
It still makes sense to factor it to a separate method with 3 args. Because it can be reused several times all over the place.
The test itself is readable, but the big picture copy/paste/duplication is not a great pattern to maintain.

Reply:
Awesome!